File: python/kode program/restoran.py
# restoran.py

from makanan import Makanan
from minuman import Minuman

class Restoran:
    def __init__(self, nama, alamat, jam_operasional):
        self.nama = nama
        self.alamat = alamat
        self.jam_operasional = jam_operasional
        self.menu_list = []

    def tambah_menu(self, menu):
        self.menu_list.append(menu)

    def tampilkan_menu(self):
        print("=" * 60)
        print(f"Selamat Datang di Restoran {self.nama}")
        print(f"Alamat: {self.alamat} | Buka: {self.jam_operasional}")
        print("=" * 60)
        
        if not self.menu_list:
            print("\nBelum ada menu yang terdaftar.")
            return

        # Header Tabel disesuaikan agar pas
        print("\n" + "-" * 136)
        print(f"| {'KODE':<5} | {'NAMA MENU':<20} | {'HARGA':>12} | {'KATEGORI':<10} | {'DETAIL':<40} | {'DESKRIPSI':<30} |")
        print("-" * 136)

        # Body Tabel
        for menu in self.menu_list:
            # Mengambil semua atribut langsung dari objek menu
            kode = menu.kode_menu
            nama = menu.nama
            harga = f"Rp {menu.harga:,}"
            deskripsi = menu.deskripsi
            
            kategori = ""
            detail = ""

            # Logika untuk membedakan Makanan dan Minuman
            if isinstance(menu, Makanan):
                kategori = "Makanan"
                detail = f"Pedas: {menu.pedas}, Jenis: {menu.jenis_masakan}"
            elif isinstance(menu, Minuman):
                kategori = "Minuman"
                detail = f"Ukuran: {menu.ukuran}, Suhu: {menu.suhu}"

            # Mencetak satu baris tabel yang sudah rapi
            print(f"| {kode:<5} | {nama:<20} | {harga:>12} | {kategori:<10} | {detail:<40} | {deskripsi:<30} |")

        # Footer Tabel
        print("-" * 136)

File: java/kode program/Restoran.java
// Restoran.java
import java.util.ArrayList;
import java.util.List;

public class Restoran {
    private String nama;
    private String alamat;
    private String jamOperasional;
    private List<Menu> menuList;

    public Restoran(String nama, String alamat, String jamOperasional) {
        this.nama = nama;
        this.alamat = alamat;
        this.jamOperasional = jamOperasional;
        this.menuList = new ArrayList<>(); // Composition & Array of Object
    }

    public void tambahMenu(Menu menu) {
        this.menuList.add(menu);
    }

    public void tampilkanMenu() {
        System.out.println("============================================================");
        System.out.printf("Selamat Datang di Restoran %s\n", this.nama);
        System.out.printf("Alamat: %s | Buka: %s\n", this.alamat, this.jamOperasional);
        System.out.println("============================================================");

        if (this.menuList.isEmpty()) {
            System.out.println("\nBelum ada menu yang terdaftar.");
            return;
        }
        
        // Header Tabel
        System.out.println("\n" + "-".repeat(145));
        System.out.printf("| %-5s | %-20s | %12s | %-10s | %-45s | %-35s |\n", "KODE", "NAMA MENU", "HARGA", "KATEGORI", "DETAIL", "DESKRIPSI");
        System.out.println("-".repeat(145));
        
        // Body Tabel
        for (Menu menu : this.menuList) {
            String kode = menu.getKodeMenu();
            String nama = menu.getNama();
            String harga = String.format("Rp %,d", menu.getHarga());
            String deskripsi = menu.getDeskripsi();
            
            String kategori = "";
            String detail = "";

            // Cek tipe objek untuk mengisi kolom Kategori dan Detail
            if (menu instanceof Makanan) {
                Makanan m = (Makanan) menu; // Type casting
                kategori = "Makanan";
                detail = String.format("Pedas: %s, Jenis: %s", m.getPedas(), m.getJenisMasakan());
            } else if (menu instanceof Minuman) {
                Minuman m = (Minuman) menu; // Type casting
                kategori = "Minuman";
                detail = String.format("Ukuran: %s, Suhu: %s", m.getUkuran(), m.getSuhu());
            }
            
            // Mencetak baris tabel
            System.out.printf("| %-5s | %-20s | %12s | %-10s | %-45s | %-35s |\n", kode, nama, harga, kategori, detail, deskripsi);
        }

        System.out.println("-".repeat(145));
    }
}